task_sdk/src/airflow/sdk/definitions/_internal/templater.py

```diff
@@ -128,6 +128,8 @@ def _do_render_template_fields(
                 setattr(parent, attr_name, rendered_content)
 
     def _render(self, template, context, dag=None) -> Any:
+        if dag and dag.is_asset_triggerable:
+            context.pop("logical_date", None)
```

Comment:
Can we not add this key to the context in the first place, instead of popping it out afterwards?

Reply:
just removed it in task_runner